app/src/main/java/com/github/se/polyfit/model/ingredient/Ingredient.kt

```diff
@@ -0,0 +1,74 @@
+package com.github.se.polyfit.model.ingredient
+
+import com.github.se.polyfit.model.meal.Meal
+import kotlin.properties.Delegates
+
+/**
+ * Represents an ingredient in a meal. Automatically updates the parent meal when any property
+ * changes.
+ *
+ * @property name is unique for each meal
+ */
+class Ingredient(
+    val name: String, // name is now a val
+    servingSize: Double,
+    calories: Double,
```

Comment:
to be discussed: we might want to store more than just the basics (calories, protein, carbs, etc), for example sub section under fat

Reply:
Currently, it is modeled based on the information provided in the documentation about the nutritional values returned. At this moment, I am not entirely sure which API to use (and what information it can return), which is why it models only simple nutritional facts for now. But I can be expanded in the future.

Reply:
The documentation sucks. You can find the list of nutriments for each meal https://github.com/swent-group10/polyfit/pull/39#discussion_r1541962229 but would you rather store all of them or a particular subset?

Reply:
Here is actually the API we're going to use. What nutritional values do you think would be a good idea to be stored?
https://spoonacular.com/food-api/docs#Nutrition-by-ID
@aprilslis